psi4/driver/driver_cbs_helper.py

```diff
@@ -475,40 +517,65 @@ def scf_xtpl_helgaker_3(functionname: str, zLO: int, valueLO: float, zMD: int, v
         np_value[nonzero_mask] -= beta * np.exp(-1 * alpha * zHI)
         np_value[~nonzero_mask] = 0.0
 
-        # Build and set from numpy routines
-        value = core.Matrix(*valueHI.shape)
-        value_view = np.asarray(value)
-        value_view[:] = np_value
-        return value
+        if verbose > 2:
+            cbsscheme = f"""\n   ==> Helgaker 3-point power SCF extrapolation for method: {functionname.upper()} <==\n"""
+            cbsscheme += f"""\n   LO-zeta ({zLO}) Data\n"""
+            cbsscheme += nppp(valueLO)
+            cbsscheme += f"""\n   MD-zeta ({zMD}) Data\n"""
+            cbsscheme += nppp(valueMD)
+            cbsscheme += f"""\n   HI-zeta ({zHI}) Data\n"""
+            cbsscheme += nppp(valueHI)
+
+            cbsscheme += f"""\n   Alpha Data\n"""
+            cbsscheme += nppp(alpha)
+            cbsscheme += f"""\n   Beta Data\n"""
+            cbsscheme += nppp(beta)
+            cbsscheme += f"""\n   Extrapolated Data\n"""
+            cbsscheme += nppp(np_value)
+            cbsscheme += "\n"
+            core.print_out(cbsscheme)
+
+        ## Build and set from numpy routines
+        #value = core.Matrix(*valueHI.shape)
+        #value_view = np.asarray(value)
+        #value_view[:] = np_value
+        #return value
+
+        np_value = core.Matrix.from_array(np_value)
+        return np_value
 
     else:
-        raise ValidationError("scf_xtpl_helgaker_3: datatype is not recognized '%s'." % type(valueLO))
+        raise ValidationError(f"scf_xtpl_helgaker_3: datatype is not recognized '{type(valueLO)}'.")
 
 
-#def corl_xtpl_helgaker_2(functionname, valueSCF, zLO, valueLO, zHI, valueHI, verbose=True):
-def corl_xtpl_helgaker_2(functionname: str, zLO: int, valueLO: float, zHI: int, valueHI: float, verbose: bool = True, alpha: float = None):
+def corl_xtpl_helgaker_2(functionname: str, zLO: int, valueLO: Extrapolatable, zHI: int, valueHI: Extrapolatable, verbose: int = 1, alpha: Optional[float] = None) -> Extrapolatable:
     r"""Extrapolation scheme for correlation energies with two adjacent zeta-level bases.
     Used by :py:func:`~psi4.cbs`.
 
     Parameters
     ----------
     functionname
-        Name of the CBS component.
+        Name of the CBS component (e.g., 'MP2') used in summary printing.
     zLO
-        Lower zeta level.
+        Zeta number of the smaller basis set in 2-point extrapolation.
     valueLO
-        Lower value used for extrapolation.
+        Energy, gradient, or Hessian value at the smaller basis set in 2-point
+        extrapolation.
     zHI
-        Higher zeta level. Should be equal to zLO + 1.
+        Zeta number of the larger basis set in 2-point extrapolation.
+        Must be `zLO + 1`.
```

Comment:
Please tell me that either this restriction or this argument is going away in DDD. Same applies for every "must be" argument.

Reply:
why ought it go away? sure it's absurdly concrete, but it's symmetric, which can be nice for user-written fns, and lets all the xtpl fns have the same-named arguments. validation occurs before any of these get called.

Reply:
If this argument is non-redundant for some extrapolation functions, I'm fine with all of them keeping it.

Reply:
I don't think the X's have to be strictly separated by one. There's work by Varandas where he optimizes the X instead of alpha. I don't think it's in scope of this PR to tackle that though.